### DISCUSSION TITLE
Help! Middleware not working

### DISCUSSION THREAD

**Discussion by l4nos** [spatie/laravel-permission#2142] [Q&A]
Hey eveyrone.

Having an issue with the middleware.

I've got an admin role which has 3 permissions

- edit_processes
- edit_core_settings
- manage_billing

Now the admin role has been assigned these permissiosn and my admin user (App\Models\User) has been correctly configured with the HasRoles and the roles and permissions load when i run the following functions.

        
       
      return $user->getAllPermissions();
      return $user->roles;

Now when I try to use the middlewares:

`        Route::prefix('processes')->middleware(['can:edit_processes'])->group(__DIR__ . '/tenant-api/process-management.php');
`

or

`        Route::prefix('processes')->middleware(['permission:edit_processes'])->group(__DIR__ . '/tenant-api/process-management.php');
`

I get an unauthorized message.

I am also using  UUIDs and have configured this correctly as per the guide.

Can someone help me please or point me in the right direction.

**Comment by PaolaRuby** [comment]
If you want help, make a simple minimal application on a laravel fresh installation
